From ae26e9dca6c7622015796676406aa974bb65c210 Mon Sep 17 00:00:00 2001
From: Gleb Kodinets <gkodinets@habana.ai>
Date: Mon, 09 Aug 2021 10:26:49 +0300
Subject: [PATCH] [SW-23529] pci: moving bar mapping related types to specs

This patch includes BAR mapping related enums and structs from
specs pci header file.

Change-Id: Id27ebfd99e748b05ac0db52002cb5d43612fe8df
---

diff --git a/lib/common/pci.c b/lib/common/pci.c
index 249142d..1774105 100644
--- a/lib/common/pci.c
+++ b/lib/common/pci.c
@@ -9,6 +9,7 @@
 #include <pcie_aux_regs.h>
 #include <common/pci_ids.h>
 #include <bitops.h>
+#include <preboot.h>
 #include "pci.h"
 
 #define IATU_BAR_NUM_OFFSET		8
diff --git a/lib/common/pci.h b/lib/common/pci.h
index 4a1665a..1082a5b 100644
--- a/lib/common/pci.h
+++ b/lib/common/pci.h
@@ -8,53 +8,10 @@
 #ifndef _PCI_H_
 #define _PCI_H_
 
-#include <preboot.h>
-
-/*
- * enum hl_pci_bar_id - pci bar number enum
- * since we are working in 64bit mode, only bars 0, 2, 4 are available.
- */
-enum hl_pci_bar_id {
-	PCI_BAR_ID_0 = 0,
-	PCI_BAR_ID_2 = 2,
-	PCI_BAR_ID_4 = 4
-};
-
-/*
- * enum hl_pci_match_mode - pci match mode per region
- * @PCI_ADDRESS_MATCH_MODE: address match mode
- * @PCI_BAR_MATCH_MODE: bar match mode
- */
-enum hl_pci_match_mode {
-	PCI_ADDRESS_MATCH_MODE,
-	PCI_BAR_MATCH_MODE
-};
-
-/*
- * struct hl_inbound_pci_region - inbound region descriptor
- * @mode: pci match mode for this region
- * @addr: region target address
- * @size: region size in bytes
- * @offset_in_bar: offset within bar (address match mode)
- * @bar: bar id
- */
-struct hl_pci_inbound_region {
-	enum hl_pci_match_mode mode;
-	u64 addr;
-	u64 size;
-	u64 offset_in_bar;
-	u8 bar;
-};
-
-/*
- * struct hl_outbound_pci_region - outbound region descriptor
- * @addr: region target address
- * @size: region size in bytes
- */
-struct hl_pci_outbound_region {
-	u64 addr;
-	u64 size;
-};
+#include <stdint.h>
+#include <stdbool.h>
+#include <types.h>
+#include <hl_pci_if.h>		/* requires stdint.h */
 
 void hl_pci_set_inbound_region(u8 region,
 			       struct hl_pci_inbound_region *pci_region);
diff --git a/lib/common/pll.h b/lib/common/pll.h
index 491635a..8acc84b 100644
--- a/lib/common/pll.h
+++ b/lib/common/pll.h
@@ -8,6 +8,10 @@
 #ifndef _HL_PLL_H_
 #define _HL_PLL_H_
 
+#include <stdint.h>
+#include <stdbool.h>
+#include <types.h>
+
 #define HL_PLL_DIV_FACTOR	0x200
 #define HL_PLL_DIV_FACTOR_CMD	0x220
 #define HL_PLL_DIV_SEL		0x280
